CONTRIBUTING.md

```diff
@@ -0,0 +1,65 @@
+# Contributing Guidelines
+
+Welcome to the **Additional Ethernet Drivers** project! We appreciate your interest in contributing. Whether you're reporting an issue, proposing a feature, or submitting a pull request (PR), your contributions are highly valued.
+
+## Submitting a Pull Request (PR)
+
+- **Fork the Repository**
+  - Start by forking the [Additional Ethernet Drivers repository](https://github.com/espressif/esp-eth-drivers) on GitHub to make your changes.
+
+- **Install Pre-commit Hooks**
+  - Set up pre-commit hooks to ensure consistent code style and linting:
+
+    ```bash
+    pip install pre-commit
+    pre-commit install-hooks
+    pre-commit install --hook-type commit-msg --hook-type pre-push
+    ```
+
+- **Commit Message Format**
+  - Follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification when writing commit messages.
+  - Commit message title format:
+    - Template: `[type]([scope]): Message`
+    - Example: `feat(ksz8863): Add port tail tagging` or `fix(lan867x): Correct datatype in lan867x_custom_ioctl`
+
+> [!TIP]
+> You can use [Commitizen](https://commitizen-tools.github.io/commitizen/) to help you format your commits correctly.
+> After installing Commitizen (see [official instructions](https://commitizen-tools.github.io/commitizen/#installation)), stage your changes and run:
+>
+> ```bash
+> cz commit
+> ```
+>
+> Then follow the prompts provided by Commitizen.
+
+- **Send a Pull Request**
+  - Submit your PR and collaborate with the maintainers through the review process. Please be patient — multiple review rounds and adjustments may be required.
+  - For quicker merging, focus your contribution on a single feature or topic. Larger or multi-topic contributions may take longer to review.
+
+## Release Process
+
+We use [Release Please](https://github.com/googleapis/release-please/tree/main) to automate the release process. It automatically generates the changelog by parsing the Git history (hence the requirement for Conventional Commit messages), bumps component versions, and creates GitHub releases and tags.
+
+To release a new version of a component, the component maintainer must:
+
+1. Ensure the build and tests pass on the latest commit.
+2. Export their GitHub token to the `GITHUB_TOKEN` environment variable.
+3. Bump the version using the provided script:
+
+   ```bash
+   ./tools/version_bump.sh
```

Comment:
The command using a relative path (`./tools/version_bump.sh`) will only work from the repo root - in subdirectories, the `./tools` folder doesn’t exist.

You may keep it as is (not a big issue).

If you want command that will work from anywhere:
```sh
$(git rev-parse --show-toplevel)/tools/version_bump.sh
```

Reply:
The point of this script is to run it even from the component folder when you want to bump only that particular component. The the script parses the path of current folder and extract component name based on it based on.
I updated the text as:
```
    ```sh
    ./tools/version_bump.sh
    ```
    or if you want to version bump just single component, navigate to the component's folder and run:
    ```sh
    ../tools/version_bump.sh
    ```
 ```